File: src/main/java/com/evanlennick/webserver/mimetype/MimeTypeUtil.java
package com.evanlennick.webserver.mimetype;

import com.google.common.collect.Maps;

import java.util.Map;

public final class MimeTypeUtil {

    private static Map<String, String> mimeTypeMap;

    private MimeTypeUtil() {}

    public static String getMimeTypeStringByFileExtension(String extension) {
        if (null == mimeTypeMap) {
            initMimeTypeMap();
        }

        String mimeType = mimeTypeMap.get(extension);
        if(null == mimeType) {
            mimeType = "application/octet-stream";
        }

        return mimeType;
    }

    private static void initMimeTypeMap() {
        mimeTypeMap = Maps.newHashMap();

        MimeType[] values = MimeType.values();
        for (MimeType value : values) {
            for (String extension : value.getExtensions()) {
                mimeTypeMap.put(extension, value.getMimeTypeString());
            }
        }
    }
}

File: src/main/java/com/evanlennick/webserver/mimetype/MimeTypeUtil.kt
package com.evanlennick.webserver.mimetype

import com.google.common.collect.Maps

object MimeTypeUtil {

    private var mimeTypeMap: MutableMap<String, String>? = null

    fun getMimeTypeStringByFileExtension(extension: String): String {
        if (null == mimeTypeMap) {
            initMimeTypeMap()
        }

        var mimeType: String? = mimeTypeMap!![extension]
        if (null == mimeType) {
            mimeType = "application/octet-stream"
        }

        return mimeType
    }

    private fun initMimeTypeMap() {
        mimeTypeMap = Maps.newHashMap<String, String>()

        val values = MimeType.values()
        for (value in values) {
            for (extension in value.extensions) {
                mimeTypeMap!!.put(extension, value.mimeTypeString)
            }
        }
    }
}
